perses/rjmc/topology_proposal.py

```diff
@@ -2144,12 +2144,18 @@ def propose(self,
 
         # Explicitly de-map any atoms involved in constrained bonds that undergo length changes
         _logger.debug(f"mapped {len(adjusted_atom_map)} before constraint repairs")
-        adjusted_atom_map = SmallMoleculeSetProposalEngine._constraint_repairs(adjusted_atom_map, current_system, new_system, current_topology, new_topology)
+        adjusted_atom_map, deleted_indices = SmallMoleculeSetProposalEngine._constraint_repairs(adjusted_atom_map, current_system, new_system, current_topology, new_topology)
         _logger.debug(f"mapped {len(adjusted_atom_map)} after constraint repairs")
+        # Also adjust small molecule atom map accordingly to deleted indices
```

Comment:
When we refactor this to use `AtomMap` objects, we shouldn't need to keep these extra utility attributes around---we can just use the repaired `AtomMap` and call `map.render_image(filename)`. But we can punt that to a future cleanup if we capture the cleanup task as an issue.

Reply:
Yes I agree, but we would still need some way to access the inner object, either to some attribute or property. But definitely, we should try to refactor this to use `AtomMapping`.

Reply:
Issue created in https://github.com/choderalab/perses/issues/1129 . Resolving conversation.